contracts/registry/src/lib.rs

```diff
@@ -1233,4 +1233,47 @@ mod tests {
         ctr.sbt_recover(alice(), bob());
         assert_eq!(ctr.sbt_supply_by_owner(bob(), issuer2(), None), 1);
     }
+
+    // sbt_ban
+    #[test]
+    fn sbt_soul_transfer_ban() {
+        let (mut ctx, mut ctr) = setup(&issuer1(), 2 * MINT_DEPOSIT);
+        let m1_1 = mk_metadata(1, Some(START + 10));
+        ctr.sbt_mint(vec![(alice(), vec![m1_1.clone()])]);
+        assert!(!ctr.is_banned(alice()));
+
+        ctx.predecessor_account_id = alice();
+        testing_env!(ctx.clone());
+        ctr.sbt_soul_transfer(alice2(), None);
+        assert!(ctr.is_banned(alice()));
+        assert!(!ctr.is_banned(alice2()));
+    }
+
+    #[test]
+    fn sbt_recover_ban() {
+        let (mut ctx, mut ctr) = setup(&issuer1(), 2 * MINT_DEPOSIT);
+        let m1_1 = mk_metadata(1, Some(START + 10));
+        ctr.sbt_mint(vec![(alice(), vec![m1_1.clone()])]);
+        assert!(!ctr.is_banned(alice()));
+
+        ctx.predecessor_account_id = issuer1();
+        testing_env!(ctx.clone());
+        ctr.sbt_recover(alice(), alice2());
+        assert!(ctr.is_banned(alice()));
+        assert!(!ctr.is_banned(alice2()));
+    }
+
+    #[test]
+    #[should_panic(expected = "account alice.near is banned")]
+    fn sbt_mint_to_banned_account() {
+        let (_, mut ctr) = setup(&issuer1(), 2 * MINT_DEPOSIT);
+        let m1_1 = mk_metadata(1, Some(START + 10));
+
+        //ban alice account
+        ctr.banlist.insert(&alice());
+        assert!(ctr.is_banned(alice()));
+
+        //try to mint to a banned account
+        ctr.sbt_mint(vec![(alice(), vec![m1_1.clone()])]);
+    }
```

Comment:
let's add more tests for soul_transfer:
1. soul_transfer from a banned account should fail
2. soul_transfer to a banned account should fail
3. soul_transfer, which requires to calls to complete (because a user has lot of tokens), bans account after the first call.

Reply:
DONE